src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1229,125 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
```

Comment:
Ah, I did that because I was trying to change the original logic as little as possible.  For instance `Backtick` returning a char `` `n`` or `z` led to further checking of that character. IOW a call to Backtick was not followed by a call to `continue` to the next loop iteration.  If you think we can always proceed onto the next iteration without further checking the returned character, then yeah I can simplify that logic.

BTW here is what I'm thinking to eliminate the duplicate code:
```c#
        private bool IsSurrogatePair(string str, ref char nonSurrogatePairChar, params StringBuilder[] addStrToStringBuilders) 
        {
            Diagnostics.Assert(!string.IsNullOrEmpty(str), "Caller never calls us with an empty string");
            Diagnostics.Assert((str.Length <= 2), "Caller never calls us with a string length greater than two");
            Diagnostics.Assert((str.Length == 1) || Char.IsSurrogate(str, 0), 
                "Caller never calls us with a string length of two and not be a surrogate pair");                
            
            if (str.Length == 1) 
            {
                nonSurrogatePairChar = str[0];
                return false;
            }

            if (addStrToStringBuilders?.Length > 0) 
            {
                foreach (var sb in addStrToStringBuilders)
                {
                    sb.Append(str);
                }
            }

            return true;
        }
```
Not crazy about the method name so I'm open to ideas but this does clean up the various call sites e.g."
```c#
                else if (c == '`')
                {
                    // If end of input, go ahead and append the backtick and issue an error later
                    char c1 = PeekChar();
                    if (c1 != 0)
                    {
                        SkipChar();
                        string str = Backtick(c1);
                        if (IsSurrogatePair(str, ref c, sb, formatSb))
                        {
                            // Skip further scanning when str is surrogate pair and is outside the basic multilingual plane
                            continue;
                        } 
                    }
                }
```